### DISCUSSION TITLE
Which version of CsqLang is supposed to be used

### DISCUSSION THREAD

**Discussion by gekkowrld** [CsqLang/Csq3#43] [Q&A]
I have looked at the org an there seems to be more than one version of the language (Csq3 and Csq4). Should I use 3 or 4?

**Comment by AniketKumar2500** [comment]
4 it's the best and easiest version.

**Comment by gekkowrld** [comment]
Is 3 still in active development (if any at all?)

**Reply by AniketKumar2500** [reply]
No it's not active because 4th version is having so many new changes which are not there in 3 so currently 4 is active in development.

**Reply by gekkowrld** [reply]
Since there is no fully detailed documentation on the features to be added or which ones are actually there, how does one determine how the development is done

**Reply by AniketKumar2500** [reply]
Basically during the development of 3rd version i was adding new features and removing bugs due to that the version became unstable which indirectly made me not to develop a documentation and finally i closed it.

**Reply by gekkowrld** [reply]
ok, I'll try to see what I can add or help develop

**Reply by AniketKumar2500** [reply]
i suggest help in development of version 4 since it will be quite easy the 3rd versions code is totally mess it doesn't create any AST which makes the management difficult but 4th has.

**Comment by gekkowrld** [comment]
Hello @AniketKumar2500 I have tried the build steps
i) clone and head to the directory
ii) run `bash build.sh`
iii) Create `hello.csq` and add `print('Hello, World!')`
iv) run ` ./csq cpp hello 'current-directory'`

Then I run into `Segmentation fault (core dumped)`
Is this expected and if yes, how do I fix it

**Reply by AniketKumar2500** [reply]
Well, it's quite weird could you please send the photograph of the code by opening it on your editor?, There is one possible reason but before I want to see the pic of it.

**Reply by gekkowrld** [reply]
Here is the process I followed. I have yet to modify any file so it is the same as the one in csqlang 4 repository.

![Screenshot of gekkowrld's terminal](https://github.com/CsqLang/Csq3/assets/108436933/bf5c40cf-88fb-4f2b-97af-c16903b39796)

**Reply by AniketKumar2500** [reply]
The issue is that while running ./csq you have to enter the current directory you are in.
Like this : ./csq cpp hello /.../CsqLang

**Reply by gekkowrld** [reply]
I found the problem, in the file `vm.h` on line 23 where it says `/Csq4/IR/instructions.h` I changed it to `IR/instructions.h` and then it worked

**Reply by AniketKumar2500** [reply]
Could you give a photo by typing ls command?

**Comment by AniketKumar2500** [comment]
And it's not any problem instead you have not used it correctly and the change you have done are quite dangerous since the logic is overruled.
In place of 'current-directory' you should place th current directory where the source file is.

**Reply by gekkowrld** [reply]
Because of how I cloned the repo, I don't have any directory known as `Csq4`. Does that count?

**Reply by AniketKumar2500** [reply]
Are you sure?

**Reply by gekkowrld** [reply]
yeah, it is a clone of my fork which for convenience sake I renamed it

**Reply by AniketKumar2500** [reply]
Ohh so now it's necessary for you to make changes according to name if name remained same so command would be same else now you should do accordingly.

**Reply by gekkowrld** [reply]
You could make the changes in the main code so no one struggles again incase they take the same approach as me
